### PR TITLE
fix(DataStore): Owner and Group Combined @auth

### PR DESCRIPTION
**Pull request by wooj2**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by wooj2** [comment]
Related:
https://github.com/aws-amplify/amplify-android/pull/860

File: AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/AuthRuleDecorator.swift [hunk 1]
```diff
@@ -42,14 +42,46 @@ public struct AuthRuleDecorator: ModelBasedGraphQLDocumentDecorator {
...
+            !authRule.groups.isEmpty &&
+            authRule.getModelOperationsOrDefault().contains(.read)
+    }
+
+    private func decorateAuthStrategy(document: SingleDirectiveGraphQLDocument,
+                                      authRule: AuthRule,
+                                      readAuthorizedGroups: Set<String>) -> SingleDirectiveGraphQLDocument {
         guard authRule.allow == .owner else {
             return document
         }
```
Comment (lawmicha) [inline AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/AuthRuleDecorator.swift:87]: can remove next guard and put it with the first
```swift
guard authRule.allow == .owner, var selectionSet = document.selectionSet else {
            return document
        }
```

**Review by lawmicha — APPROVED** [review]
approved with suggestions, feel free to take what you like

File: AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/AuthRuleDecorator.swift [hunk 2]
```diff
@@ -42,14 +42,46 @@ public struct AuthRuleDecorator: ModelBasedGraphQLDocumentDecorator {
...
+            }
+        }
+        return readAuthorizedGroups
+    }
+
+    private func isReadRestrictingStaticGroup(_ authRule: AuthRule) -> Bool {
+        return authRule.allow == .groups &&
+            !authRule.groups.isEmpty &&
+            authRule.getModelOperationsOrDefault().contains(.read)
+    }
```
[the remark this answers is not in the document]
Comment (wooj2) [reply AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/AuthRuleDecorator.swift:80]: NICE!!  i moved a couple other functions over as well for consistency.

**Pull request merged**